docs/guide/coordinates/index.rst

```diff
@@ -0,0 +1,324 @@
+  >>> hpc_out = sunpy.coordinates.Helioprojective(observer="venus", obstime="2017-07-26")
+
+  Perform the transformation from one to the other.
+  >>> hpc2 = hpc1.transform_to(hpc_out)
+
+An example with two maps, named ``aia`` and ``stereo``::
+
+  >>> hpc1 = SkyCoord(0*u.arcsec, 0*u.arcsec, frame=aia.coordinate_frame)  # doctest: +SKIP
+  >>> hpc2 = hpc1.transform_to(stereo.coordinate_frame)  # doctest: +SKIP
+
+
+Design of the Coordinates Sub-Package
+=====================================
+
+This sub-package works by defining a collection of ``Frames``
+(`sunpy.coordinates.frames`), which exists on a transformation graph, where the
+transformations between the coordinate frames are then defined and registered
+with the transformation graph (`sunpy.coordinates.transformations`). It is also
+possible to transform SunPy frames to Astropy frames.
+
+Positions within these ``Frames`` are stored as a ``Representation`` of a
+coordinate, a representation being a description of a point in a Cartesian,
+spherical or cylindrical system (see :ref:`astropy-coordinates-representations`). A frame
+that contains a representation of one or many points is said to have been
+'realized'.
+
+For a more in depth look at the design and concepts of the Astropy coordinates
+system see :ref:`astropy-coordinates-overview`
+
+
+
+Frames and SkyCoord
+-------------------
+
+The `~astropy.coordinates.SkyCoord` class is a high level wrapper around the
+`astropy.coordinates` sub-package. It provides an easier way to create and transform
+coordinates, by using string representations for frames rather than the classes
+themselves and some other usability improvements, for more information see the
+`~astropy.coordinates.SkyCoord` documentation.
+
+The main advantage provided by `~astropy.coordinates.SkyCoord` is the support it
+provides for caching Frame attributes. Frame attributes are extra data specified
+with a frame, some examples in `sunpy.coordinates` are ``obstime`` or
+``observer`` for observer location. Only the frames where this data is
+meaningful have these attributes, i.e. only the Helioprojective frames have
+``observer``. However, when you transform into another frame and then back to a
+projective frame using `~astropy.coordinates.SkyCoord` it will remember the attributes previously
+provided, and repopulate the final frame with them. If you were to do
+transformations using the Frames alone this would not happen.
+
+The most important implication for this in `sunpy.coordinates` is the ``rsun``
+parameter in the projective frames. If you create a projective frame with a
+``rsun`` attribute, if you convert back to a projective frame it will be set
+correctly. It should also be noted that, if you create a Heliographic frame and
+then transform to a projective frame with an ``rsun`` attribute, it will not
+match the ``radius`` coordinate in the Heliographic frame. This is because you may
+mean to be describing a point above the defined 'surface' of the Sun.
```

Comment:
```suggestion
```
This paragraph is obsolete because `HeliographicStonyhurst` has had a `rsun` frame attribute for quite a while now.